osu.Desktop/DiscordRichPresence.cs

```diff
@@ -173,6 +192,41 @@ private string truncate(string str)
             return null;
         }
 
+        private async void fetchBeatmapSet()
```

Comment:
Use `async Task` instead of `async void`.

Reply:
`fetchBeatmapSet()` is being called from within a `BindValueChanged()` callback which is fundamentally incompatible with `async`. So while yes, `async void` is bad, it's going to have to happen _somewhere_.

The best that can be done here is probably a continuation on top of this to handle failures.